### PR TITLE
Sasb 2245 support external

### PR DESCRIPTION
**Pull request by superlama**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -133,6 +133,11 @@
             <version>4.5.10</version>
             <scope>compile</scope>
         </dependency>
+        <dependency>
```
Comment (aetherknight) [inline pom.xml:136]: It won't let me comment on it above, but is easymock a runtime dependency or a test dependency?

File: pom.xml [hunk 2]
```diff
@@ -133,6 +133,11 @@
...
+          <groupId>org.apache.commons</groupId>
+          <artifactId>commons-lang3</artifactId>
+          <version>3.9</version>
```
Comment (aetherknight) [inline pom.xml:139]: I'm not as familiar with the details of Java package/dependency management:

Will dependencies in the POM file impact installation of the plugin (or is this just for development)? Or are we using libraries that are standard on Jenkins installs?

File: src/main/java/com/datatheorem/mobileappsecurity/jenkins/plugin/SendBuildToDataTheoremPublisher.java [hunk 3]
```diff
@@ -186,7 +186,13 @@ public void perform(
             }
             sendBuild.setReleaseType(releaseType);
         }
-        if (applicationCredentialUsername != null  && !applicationCredentialUsername.isEmpty()) {
+
+        if (externalId != null && !externalId.isEmpty()) {
```
Comment (aetherknight) [inline src/main/java/com/datatheorem/mobileappsecurity/jenkins/plugin/SendBuildToDataTheoremPublisher.java:190]: 👍

File: pom.xml (hunk 1 above)
Comment (marc-tranzer) [reply pom.xml:136]: can you change the easymock dependency scope to test

**Review by marc-tranzer — APPROVED** [review]
sounds good. Added one request

**Pull request merged**